File: skellig-connection-jdbc/src/main/java/org/skellig/connection/jdbc/model/JdbcDetails.java
package org.skellig.connection.database.model;

public class JdbcDetails {

    private String driverName;
    private String url;
    private String userName;
    private String password;


    public JdbcDetails(String driverName, String url, String userName, String password) {
        this.driverName = driverName;
        this.url = url;
        this.userName = userName;
        this.password = password;
    }

    public String getDriverName() {
        return driverName;
    }

    public String getUrl() {
        return url;
    }

    public String getUserName() {
        return userName;
    }

    public String getPassword() {
        return password;
    }
}

File: skellig-test-step-processing-jdbc/src/main/kotlin/org/skellig/teststep/processor/jdbc/model/JdbcDetails.kt
package org.skellig.teststep.processor.jdbc.model

import org.skellig.teststep.processor.db.model.DatabaseDetails


/**
 * Represents the details of a JDBC connection.
 * It extends the [DatabaseDetails] class and adds the driver name and JDBC URL.
 *
 * @param serverName The name of the database server.
 * @param driverName The name of the JDBC driver.
 * @param url The JDBC URL to connect to the database server.
 * @param userName The username to connect to the database server. Can be null.
 * @param password The password to connect to the database server. Can be null.
 */
class JdbcDetails(
    serverName: String,
    val driverName: String,
    val url: String,
    userName: String?,
    password: String?
) : DatabaseDetails(serverName, userName, password) {

    override fun toString(): String {
        return "(${super.toString()}, driverName = $driverName, url = $url)"
    }
}